opencog/cython/PythonEval.cc

```diff
@@ -44,60 +52,239 @@ using namespace opencog;
 //#define DPRINTF printf
 #define DPRINTF(...)
 
+
+static const char* DEFAULT_PYTHON_MODULE_PATHS[] =
+{
+    PROJECT_BINARY_DIR"/opencog/cython", // bindings
+    PROJECT_SOURCE_DIR"/opencog/python", // opencog modules written in python
+    PROJECT_SOURCE_DIR"/tests/cython",   // for testing
+    DATADIR"/python",                    // install directory
+    #ifndef WIN32
+    "/usr/local/share/opencog/python",
+    "/usr/share/opencog/python",
+    #endif // !WIN32
+    NULL
+};
+
+
 PythonEval* PythonEval::singletonInstance = NULL;
+AtomSpace* PythonEval::singletonAtomSpace = NULL;
```

Comment:
Well, right now Shujing's pattern miner relies on there being 2 Atomspaces
working togeter

So, if the python interface assumes only one Atomspace, then the python
interface can't be used to work with the pattern miner

This seems undesirable, thus I suggest the python interface should not
assume a singleton Atomspace

-- Ben

On Sat, Feb 28, 2015 at 12:40 PM, Linas Vepštas <notifications@github.com>
wrote:

> In opencog/cython/PythonEval.cc
> <https://github.com/opencog/opencog/pull/1325#discussion_r25554330>:
>
> >  PythonEval* PythonEval::singletonInstance = NULL;
> > +AtomSpace* PythonEval::singletonAtomSpace = NULL;
>
> again, in the long-run, the atomspace can't be a singleton, although,
> short-term, that's OK-ish. Other atomspaces are expected to hold
> temporaries of various sorts, or to hold special-purpose subsets of the
> data to avoid clutter. Its still rather experimental, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1325/files#r25554330>.
>



-- 
Ben Goertzel, PhD
http://goertzel.org

"The reasonable man adapts himself to the world: the unreasonable one
persists in trying to adapt the world to himself. Therefore all progress
depends on the unreasonable man." -- George Bernard Shaw